htlcswitch/link.go

```diff
@@ -610,6 +612,20 @@ func (l *channelLink) handleDownStreamPkt(pkt *htlcPacket, isReProcess bool) {
 			"local_log_index=%v, batch_size=%v",
 			htlc.PaymentHash[:], index, l.batchCounter+1)
 
+		// If packet was forwarded from another channel link then we should
```

Comment:
I think this should instead be done once the HTLC has actually been _locked into_ the commitment transaction. This isn't absolutely necessary now, but seems like a requirement for your switch persistent ideas to be implemented properly. In order to do this, we'd need to modify the `ReceiveRevocation` method to also scan _our_ update log and return any locked-in entries. Doing this would be independently useful as it would add another synchronization hook for many unit tests in the project that currently jut sleep as they wait for an HTLC to be locked in.

Reply:
I agree that is a better time to add to the circuit map. Another complication is that we'll need to add `sourceChannelID` and `sourceHTLCID` to PaymentDescriptor so that we know what to put in the circuit map after they revoke. That is part of my master plan in terms of persistence and stuff though, so I'll go ahead with it and we can see how it looks.